### PR TITLE
[ALLUXIO-2518] Adding REST service for fetching proxy information

### PR DESCRIPTION
**Pull request by jsimsa**
https://alluxio.atlassian.net/browse/ALLUXIO-2518

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/DefaultAlluxioMaster.java [hunk 1]
```diff
@@ -244,14 +244,12 @@ public void start() throws Exception {
 
   @Override
   public void stop() throws Exception {
+    LOG.info("Stopping Alluxio master @ {}", mRpcAddress);
```
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/DefaultAlluxioMaster.java]: Should this still be in the `if` clause?

File: core/server/src/main/java/alluxio/proxy/AlluxioProxy.java [hunk 2]
```diff
@@ -58,66 +52,9 @@ public static void main(String[] args) {
...
-      LOG.error("Uncaught exception while running Alluxio proxy, stopping it and exiting.", e);
-      try {
-        proxy.stop();
-      } catch (Exception e2) {
-        // continue to exit
-        LOG.error("Uncaught exception while stopping Alluxio proxy, simply exiting.", e2);
-      }
-      System.exit(-1);
-    }
+    AlluxioProxyService proxy = new DefaultAlluxioProxy();
```
Comment (calvinjia) [inline core/server/src/main/java/alluxio/proxy/AlluxioProxy.java]: Should we use the factory here?

File: core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java [hunk 3]
```diff
@@ -214,13 +214,11 @@ public void start() throws Exception {
 
   @Override
   public void stop() throws Exception {
+    LOG.info("Stopping Alluxio worker @ {}", mRpcAddress);
```
Comment (calvinjia) [inline core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java]: Should this be in the `if` still?

File: minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java [hunk 4]
```diff
@@ -70,7 +71,7 @@
    * @param numWorkers the number of workers to run
    */
   public AbstractLocalAlluxioCluster(int numWorkers) {
-    mProxy = new AlluxioProxy();
+    mProxy = new DefaultAlluxioProxy();
```
Comment (calvinjia) [inline minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java]: Should we use the Factory here?

File: core/server/src/main/java/alluxio/master/DefaultAlluxioMaster.java (hunk 1 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/DefaultAlluxioMaster.java]: we previously had similar message in both branches, so I factored it out

File: core/server/src/main/java/alluxio/proxy/AlluxioProxy.java (hunk 2 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/proxy/AlluxioProxy.java]: done

File: core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java (hunk 3 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java]: similar response as for Alluxio master

File: minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java (hunk 4 above)
Comment (jsimsa) [inline minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java]: done

**Comment by jsimsa** [comment]
@calvinjia addressed your comments, thanks

**Comment by jsimsa** [comment]
@calvinjia PTAL

**Pull request merged**